include/debug/gdbstub.h

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2020 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEBUG_GDBSTUB_H_
+#define ZEPHYR_INCLUDE_DEBUG_GDBSTUB_H_
+
+#include <arch/gdbstub.h>
+
+/**
+ * @brief Start debugging
+ *
+ * This function will stop Zephyr execution and wait for a gdb
+ * connection.  Once gdb is connected it will loop waiting for a
+ * instruction.
+ *
+ * @note Just call this function when the selected backend was
+ * properly initialized. Be aware that different backends can have
+ * different initialization requirements.
+ */
+void z_gdb_start(void);
```

Comment:
this doesn't appear to be an application-facing API.
Can we define in kernel/include/ instead?

Reply:
Maybe, I was not sure where starting to debug. Initially I let it to the application start debug itself. My doubt is because different backends can have different initialization requirements. Don't know if there is a case that needs to expose it though,

Reply:
Looking it again, other debug functionalities live out of the kernel, like coverage and tracing. IMHO is more consistent keep this way. Will remove the `z_` namespace though.